Source/JavaScriptCore/runtime/AtomicsObject.cpp

```diff
@@ -522,9 +626,17 @@ JSC_DEFINE_HOST_FUNCTION(atomicsFuncNotify, (JSGlobalObject* globalObject, CallF
     if (!typedArrayView->isShared())
         return JSValue::encode(jsNumber(0));
 
+    JSPromise* promise = typedArrayView->possiblySharedBuffer()->getPromise(accessIndex);
+    dataLogF("Thread <%p> ArrayBuffer* addr: %p\n", &Thread::current(),typedArrayView->possiblySharedBuffer());
+    if (promise) {
+        JSGlobalObject* global = promise->globalObject();
+        promise->resolve(global, jsNontrivialString(global->vm(), "ok"_s));
+    }
+
```

Comment:
`typedArrayView->possiblySharedBuffer()` in `Notify` provides different `ArrayBuffer*` from `waitAsync`.